src/node/internal/process.ts

```diff
@@ -20,13 +20,90 @@ export function nextTick(cb: Function, ...args: unknown[]) {
   });
 }
 
-// Note that there is no process-level environment in workers so the process.env
-// object is initially empty. This is different from Node.js where process.env
-// picks up values from the operating system environment. The configured bindings
-// for the worker are accessible from the env argument passed into the fetch
-// handler and have no impact here.
+// Decide if a value can round-trip to JSON without losing any information.
+function isJsonSerializable(value: any): boolean {
+  switch (typeof value) {
+    case "boolean":
+    case "number":
+    case "string":
+      return true;
 
-export const env = new Proxy(utilImpl.getEnvObject(), {
+    case "object": {
+      if (value === null) {
+        return true;
+      }
+
+      // We only consider objects to be serializable if they are plain objects or plain arrays.
+      // Any subclass is not serializable because serializing it would be lossy.
+      //
+      // (Practically speaking, we don't want, say, a KV namespace to show up in process.env as
+      // "{}" -- this would be weird.)
+      switch (Object.getPrototypeOf(value)) {
```

Comment:
An object with a null prototype is also safe to json stringify

Reply:
Technically objects with _any_ prototype are safe to JSON-stringify. What this function is testing for is whether the object looks like it came from JSON / can round-trip to JSON without losing its type.

Reply:
Sorry, I think you didn't understand my comment. I do not want to permit null prototypes here, for the same reason that I don't want to permit prototypes of arbitrary subclasses of Object: they would be lost in a round-trip, therefore the type is not really JSON-serializable.

Reply:
I would suggest adding a comment then to that effect since a null-prototype object is technically JSON stringify-able. Something like, "Ignore null prototype objects because, while they can be JSON stringified, they cannot be parsed with a null prototype".

Reply:
heh, just saw the updated comment :-) ... so nevermind! :-)